eslint/no-relative-parent-import.mjs

```diff
@@ -5,7 +5,7 @@ import * as posix from 'node:path/posix';
 
 import { ESLintUtils } from '@typescript-eslint/utils';
 
-const projectRoot = posix.normalize(path.dirname(import.meta.dirname));
+const projectRoot = posix.normalize(path.dirname(import.meta.dirname || '.'));
```

Comment:
为啥要改这里？

Reply:
因为本地跑 lint 会报错这个是 undefined，没法 commit

Reply:
你咋跑的lint？`pnpm run lint`?

Reply:
哦我知道了，你是不是用的node18？

Reply:
node20 呀

Reply:
那不会呀？

```text
❯❯ ~\..\..\server-private git:(master) code .
❯❯ ~\..\..\server-private git:(master) git add .
❯❯ ~\..\..\server-private git:(master) git cm test

> app@0.8.0 lint-staged C:\Users\Trim21\proj\bangumi\server-private
> lint-staged

✔ Preparing lint-staged...
✔ Running tasks for staged files...
✖ Prevented an empty git commit!
✔ Reverting to original state because of errors...
✔ Cleaning up temporary files...

  ⚠ lint-staged prevented an empty git commit.
  Use the --allow-empty option to continue, or check your task configuration

 ELIFECYCLE  Command failed with exit code 1.
husky - pre-commit script failed (code 1)
❯❯ ~\..\..\server-private git:(master) node --version
v20.13.1
```

Reply:
```shell
❯ pnpm lint   

> app@0.8.0 lint /Users/everpcpc/src/bangumi-server-private
> eslint -c eslint.config.mjs ./


Oops! Something went wrong! :(

ESLint: 9.11.1

TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
    at validateString (node:internal/validators:162:11)
    at Module.dirname (node:path:1279:5)
    at file:///Users/everpcpc/src/bangumi-server-private/eslint/no-relative-parent-import.mjs:8:42
    at ModuleJob.run (node:internal/modules/esm/module_job:218:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:329:24)
    at async importModuleDynamicallyWrapper (node:internal/vm/module:431:15)
    at async loadFlatConfigFile (/Users/everpcpc/src/bangumi-server-private/node_modules/.pnpm/eslint@9.11.1_jiti@1.21.6/node_modules/eslint/lib/eslint/eslint.js:390:21)
    at async calculateConfigArray (/Users/everpcpc/src/bangumi-server-private/node_modules/.pnpm/eslint@9.11.1_jiti@1.21.6/node_modules/eslint/lib/eslint/eslint.js:473:28)
    at async ESLint.lintFiles (/Users/everpcpc/src/bangumi-server-private/node_modules/.pnpm/eslint@9.11.1_jiti@1.21.6/node_modules/eslint/lib/eslint/eslint.js:922:25)
    at async Object.execute (/Users/everpcpc/src/bangumi-server-private/node_modules/.pnpm/eslint@9.11.1_jiti@1.21.6/node_modules/eslint/lib/cli.js:502:23)
 ELIFECYCLE  Command failed with exit code 2.
```

```shell
❯ node --version
v20.10.0
```

Reply:
最近是有什么依赖变动吗，之前都是没问题的，今天 merge 了一下 main 就不行了

Reply:
你 `pnpm i` 一下？

Reply:
```shell
❯ pnpm i
Lockfile is up to date, resolution step is skipped
Already up to date

> app@0.8.0 prepare /Users/everpcpc/src/bangumi-server-private
> husky

Done in 2.1s
```

Reply:
之前不能 commit 我还删了 `node_modules` 重新 install 了一下，装完还是不行

Reply:
神秘

Reply:
改成这个吧

```ts
import { dirname } from 'node:path';
import { fileURLToPath } from 'node:url';
    
const __dirname = dirname(fileURLToPath(import.meta.url));
```

Reply:
> 之前不能 commit 我还删了 `node_modules` 重新 install 了一下，装完还是不行

我知道了，要 node>=20.11

Reply:
要不你升级下node吧，别改这个了（）

Reply:
也行吧，我升一下看看

Reply:
好了。好奇怪之前为啥没问题